app/routes.py

```diff
@@ -1,2 +1,174 @@
+            "channel": "task_notifications",
+            "text": f"Someone  just completed the task {task.title}"
+            }
+        
+        requests.post(PATH, data=params)
+        return jsonify({"task": task.to_dict()}), 200
+
+@task_bp.route("/<task_id>/mark_incomplete", methods =["PATCH"])
+def incompleted_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        abort(404)
+    else:
+        task.completed_at = None
+        db.session.commit()
+        return jsonify({"task": task.to_dict()}), 200
+
+# GOALS
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_data = request.get_json()
+
+    if "title" not in request_data :
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_goal = Goal(title = request_data["title"])
+#connecting to the db
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_dict()}), 201
+
+@goal_bp.route("", methods = ["GET"])
+def get_goals():
+    goal_response = []
+    if request.args.get("sort")=="asc":
+        goals=Goal.query.order_by(Goal.title.asc())
+    elif request.args.get("sort")=="desc":
+        goal_response=Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+    for goal in goals:
+        goal_response.append(goal.to_dict())
+    return jsonify(goal_response), 200
+
+@goal_bp.route("/<goal_id>", methods = ["GET", "PUT"])
+def get_goal(goal_id):
+    goal_id = int(goal_id)
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return make_response("", 404)
+    if request.method == "GET":
+        return jsonify({"goal": goal.to_dict()}), 200
+    elif request.method == "PUT":
+        request_data = request.get_json()
+        goal.title = request_data["title"]
+        db.session.commit()
+        return jsonify({"goal": goal.to_dict()}), 200
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        db.session.delete(goal)
+        db.session.commit()
+        return {"details": f'Goal {goal_id} "{goal.title}" successfully deleted'}, 200
+
+    else: 
+        abort(404)
+
+#Appending a task column to the goal table
+@goal_bp.route("/<goal_id>/tasks", methods = ["POST"])
+def task_ids_to_goal(goal_id):
+    #Returning an instance of Goal
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response(""), 404
+    
+    request_data = request.get_json()
+    task_ids = []
+    for task_id in request_data["task_ids"]:
+        task = Task.query.get(task_id)
+        goal.tasks.append(task)
+        task_ids.append(task_id)
+    db.session.commit()
+    return jsonify({"id": goal.goal_id, "task_ids": task_ids}), 200
+
+@goal_bp.route("/<goal_id>/tasks", methods = ["GET"])
+def get_tasks_for_specific_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response(""), 404
```

Comment:
Just a chris-ism:  I think it's helpful to include a message about error responses in addition to the response code.